src/node_file.cc

```diff
@@ -95,11 +97,74 @@ class FSReqWrap: public ReqWrap<uv_fs_t> {
   DISALLOW_COPY_AND_ASSIGN(FSReqWrap);
 };
 
+class Path {
+  public:
+    explicit Path(v8::Isolate* isolate, v8::Local<v8::Value> value);
+
+    ~Path() {
+      if (str_ != str_st_)
+        free(str_);
```

Comment:
You switched to using `new char` below. So this should be switched to use `delete str_`.